### PR TITLE
Support Py3.11

### LINKED ISSUE #2248: Support for python 3.11
3.11 betas are coming out, with this version having some breakage in the C API.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #2248

### CODE REVIEW

**Comment by benjeffery** [comment]
Currently failing due to a lack of h5py wheels for 3.11.

File: python/requirements/CI-complete/requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ msgpack==1.0.4
 msprime==1.2.0
 networkx==2.6.3 # Held at 2.6.3 for Python 3.7 compatibility
 newick==1.3.2
-numpy==1.21.6 # Held at 1.21.6 for Python 3.7 compatibility
+numpy # Held at 1.21.6 for Python 3.7 compatibility
```
Comment (jeromekelleher) [inline python/requirements/CI-complete/requirements.txt:14]: Comment no longer correct?

File: python/requirements/development.txt [hunk 2]
```diff
@@ -10,7 +10,7 @@ h5py>=2.6.0
 jsonschema>=3.0.0
 jupyter-book>=0.12.1
 kastore
-lshmm
+#lshmm
```
Comment (jeromekelleher) [inline python/requirements/development.txt:13]: We should probably still keep this in the development requirements?

File: python/requirements/CI-complete/requirements.txt (hunk 1 above)
Comment (benjeffery) [reply python/requirements/CI-complete/requirements.txt]: Accidental change! Fixed

File: python/requirements/development.txt (hunk 2 above)
Comment (benjeffery) [reply python/requirements/development.txt]: Yep, reinstated.

**Comment by benjeffery** [comment]
Fixed the h5py issue, and a new CI issue with the docs. Now codecov is flakey so will re-run and merge tomorrow!

**Comment by benjeffery** [comment]
Manual merge due to mergify config change.

**Pull request merged**